modules/apps/frontend-view-state/frontend-view-state-service/src/main/java/com/liferay/frontend/view/state/internal/jaxrs/application/FrontendViewStateApplication.java

```diff
@@ -0,0 +1,147 @@
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.view.state.internal.jaxrs.application;
+
+import com.liferay.frontend.view.state.model.FVSActiveEntry;
+import com.liferay.frontend.view.state.model.FVSEntry;
+import com.liferay.frontend.view.state.service.FVSActiveEntryLocalService;
+import com.liferay.frontend.view.state.service.FVSEntryLocalService;
+import com.liferay.portal.kernel.json.JSONFactory;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.model.User;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Application;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.jaxrs.whiteboard.JaxrsWhiteboardConstants;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+@Component(
+	property = {
+		JaxrsWhiteboardConstants.JAX_RS_APPLICATION_BASE + "=/frontend-view-state/app",
+		JaxrsWhiteboardConstants.JAX_RS_NAME + "=Liferay.Frontend.View.State",
+		"auth.verifier.auth.verifier.PortalSessionAuthVerifier.urls.includes=/*",
+		"auth.verifier.guest.allowed=true", "liferay.oauth2=false"
+	},
+	service = Application.class
+)
+public class FrontendViewStateApplication {
+
+	@GET
+	@Path("/active-view/{clayDataSetDisplayId}")
+	@Produces(MediaType.APPLICATION_JSON)
+	public Response getActiveView(
+		@PathParam("clayDataSetDisplayId") String clayDataSetDisplayId,
+		@QueryParam("plid") long plid,
+		@QueryParam("portletId") String portletId, @Context User user) {
+
+		try {
+			FVSActiveEntry fvsActiveEntry =
+				_fvsActiveEntryLocalService.fetchFVSActiveEntry(
+					user.getUserId(), clayDataSetDisplayId, plid, portletId);
+
+			if (fvsActiveEntry == null) {
+				return Response.status(
+					Response.Status.NOT_FOUND
+				).build();
+			}
+
+			FVSEntry fvsEntry = _fvsEntryLocalService.getFVSEntry(
+				fvsActiveEntry.getFvsEntryId());
+
+			return Response.ok(
+				_jsonFactory.createJSONObject(fvsEntry.getViewState()),
+				MediaType.APPLICATION_JSON
+			).build();
+		}
+		catch (Exception exception) {
+			_log.error("Unable to get active view", exception);
+		}
+
+		return Response.status(
+			Response.Status.NOT_FOUND
```

Comment:
I think this should be a 500

Reply:
And maybe* we should just bubble up the exception and capture it with a global exception mapper that returns 500. I think there is not a default one (vulcan registers one but being outside rest builder, better not mix)

Reply:
Again, this was copied from [FrontendClayApplication](https://github.com/liferay/liferay-portal/blob/26c6bc1b470b21c0ea00632554536f3a9ad8e17b/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/internal/jaxrs/application/FrontendClayApplication.java#L115-L121), but if we want to make it better (SERVER_ERROR makes more sense than NOT_FOUND), we can change it.

> maybe* we should just bubble up the exception and capture it with a global exception mapper that returns 500

What happens now on unhandled exceptions? I expect Jackson to return a 500 anyway, as Tomcat does... :thinking: If we don't know, I can check it.

Reply:
The JAX-RS specification say that you should return a 500. But our implementation has a [bug](https://issues.liferay.com/browse/LPS-131519) in some situations (a Fault launched does not get processed). So a global `ExceptionMapper` is needed that catches all exceptions and returns a 500.

Reply:
I will go with a global ExceptionMapper to clean the code and remove the catch block and final return command